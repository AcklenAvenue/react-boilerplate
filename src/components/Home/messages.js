import { defineMessages } from 'react-intl';

const messages = defineMessages({
  GetStarted: {
    id: 'App.Home.GetStarted',
    defaultMessage: 'To get started, edit {file} and save to reload.'
  }
});

export default messages;
