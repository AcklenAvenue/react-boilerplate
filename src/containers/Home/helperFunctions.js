// $FlowFixMe
import { equals } from 'ramda';

// $FlowFixMe
const isCheckbox = id => {
  return equals(id, 'MessagePrivacy');
};

// $FlowFixMe
const isTitle = id => {
  return equals(id, 'MessageTitle');
};

// $FlowFixMe
const isContent = id => {
  return equals(id, 'MessageContent');
};

export { isCheckbox, isTitle, isContent };
