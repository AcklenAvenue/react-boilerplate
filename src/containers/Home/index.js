import React, { Component } from 'react';
import ListOfMessages from '../../components/ListOfMessages';
import SendMessage from '../../components/SendMessage';
class Home extends Component<any, any> {
  state = {
    mensajes: [
      {
        title: 'primer Mensaje',
        mensaje: 'este es un mensaje',
        public: true
      },
      {
        title: 'segundo Mensaje',
        mensaje: 'este es un mensaje',
        public: false
      }
    ]
  };

  render() {
    return (
      <div>
        <ListOfMessages Messages={this.state.mensajes} />
        <SendMessage />
      </div>
    );
  }
}

export default Home;
