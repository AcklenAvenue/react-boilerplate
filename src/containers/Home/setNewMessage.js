// $FlowFixMe
const newTitle = (state, newTitles) => {
  return { ...state, title: newTitles };
};

// $FlowFixMe
const newContent = (state, newContents) => {
  return { ...state, mensaje: newContents };
};

// $FlowFixMe
const newPrivacy = (state, newPrivacy) => {
  return { ...state, public: newPrivacy };
};
export { newTitle, newContent, newPrivacy };
