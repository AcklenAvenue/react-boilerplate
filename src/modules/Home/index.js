// @flow

import React, {Component} from 'react';
import {Jumbotron} from 'react-bootstrap';

export class Home extends Component<void> {
  render() {
    return (
      <Jumbotron>
        <p className="App-intro">
          To get started, edit <code>src/App.js</code> and save to reload.
        </p>
      </Jumbotron>
    );
  }
}

export default Home;
