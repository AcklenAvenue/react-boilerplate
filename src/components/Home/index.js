// @flow

import React from 'react';
import { Jumbotron } from 'react-bootstrap';
import styled from 'styled-components';
import { FormattedMessage } from 'react-intl';
import messages from './messages';

const AppIntro = styled.p`font-size: large;`;

const Home = () => (
  <Jumbotron>
    <AppIntro>
      <FormattedMessage {...messages.GetStarted} values={{ file: <code>src/App.js</code> }} />
    </AppIntro>
  </Jumbotron>
);

export default Home;
