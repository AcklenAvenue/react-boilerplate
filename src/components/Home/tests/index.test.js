import React from 'react';
import Home from '../index';
import renderer from 'react-test-renderer';
import { IntlProvider } from 'react-intl';

describe('<Home />', () => {
  it('Expect to be render correctly', () => {
    const wrapper = renderer.create(
      <IntlProvider locale={'en'}>
        <Home />
      </IntlProvider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
