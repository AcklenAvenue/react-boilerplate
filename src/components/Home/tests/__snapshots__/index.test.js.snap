// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> Expect to be render correctly 1`] = `
<div
  className="jumbotron"
>
  <p
    className="sc-bdVaJa dQSkNS"
  >
    To get started, edit 
    <code>
      src/App.js
    </code>
     and save to reload.
  </p>
</div>
`;
